### ISSUE TITLE
prod for tuples uses *

### ISSUE THREAD

**Issue by yurivish** [JuliaLang/julia#39183]
```julia
julia> prod([Int8(100), Int8(100)])
10000

julia> prod((Int8(100), Int8(100)))
16

julia> versioninfo()
Julia Version 1.5.3
Commit 788b2c77c1 (2020-11-09 13:37 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
```

Ref. https://github.com/JuliaLang/julia/issues/39182, which is the same issue but for `sum` and `+`

The is a todo in the code that says this function definition is inconsistent with regular `prod`:
https://github.com/JuliaLang/julia/blob/788b2c77c10c2160f4794a4d4b6b81a95a90940c/base/tuple.jl#L399-L400

**Comment by yurivish** [comment]
I just noticed that the `sum` and `prod` definitions are right next to each other with the same comment on both, so it might not be worth having a separate issue. Would the fix be the same in both cases?

**Comment by NHDaly** [comment]
👍 according to the docstring, this is definitely a bug, yeah:
```julia
help?> prod([Int8(100), Int8(100)])
  prod(itr)

  Returns the product of all elements of a collection.

  The return type is Int for signed integers of less than system word size, and UInt for unsigned integers of less than system word size. For all other
  arguments, a common return type is found to which all arguments are promoted.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> prod(1:20)
  2432902008176640000
```

**Comment by NHDaly** [comment]
Just to confirm, indeed the docstring says this is supposed to be promoting to an Int:
```julia
 sum(itr)

  Returns the sum of all elements in a collection.

  The return type is Int for signed integers of less than system word size, and UInt for unsigned integers of less than system word size. For all other
  arguments, a common return type is found to which all arguments are promoted.
```

**Comment by NHDaly** [comment]
Closed in #41510.
